numpy/lib/function_base.py

```diff
@@ -743,40 +742,18 @@ def histogram(a, bins=10, range=None, normed=False, weights=None,
             else:
                 tmp_w = weights[i:i + BLOCK]
 
-            # Only include values in the right range
             keep = (tmp_a >= mn)
```

Comment:
don't know how relevant this is, but can't this also be moved into the C code?

Reply:
I left this out originally, but found that when the histogram covers only a small part of the sample range (10% or less), adding this back made the calculation much faster (well more than a 2x speed up) and it did not change the timing much for when the histogram range was wider than the sample range. (I'm in the process of adding benchmarks for these cases.)